bpaotu/bpaotu/views.py

```diff
@@ -593,3 +613,115 @@ def dev_only_ckan_check_permissions(request):
     response = '||'.join([digest, data])
 
     return HttpResponse(response)
+
+
+# --------------------------------------------------------------------------------
+# Add Images to Map
+# --------------------------------------------------------------------------------
+
+# img file extension corresponding to pillow processing format and HttpResponse format
+
+IMG_EXTENSION_TABLE = {
+    'jpg': ['JPEG', 'image/jpeg']
+}
+
+
+LOOKUP_TABLE_KEY = 'lkup_tbl_key'
+
+
+def _get_cached_item(key):
+    try:
+        cache = caches['image_results']
+
+        return cache.get(key)
+    except Exception as e:
+        logger.error(str(e))
+
+
+def _set_cached_item(key, value):
+    try:
+        cache = caches['image_results']
+        cache.set(key, value, CACHE_7DAYS)
+    except Exception as e:
+        logger.error(str(e))
+
+    return True
+
+
+def _create_img_lookup_table():
+    '''
+    Create a lookup table of all images in ckan.
+    '''
+    if _get_cached_item(LOOKUP_TABLE_KEY) is None:
+        remote = ckanapi.RemoteCKAN(settings.BPA_PROD_URL, apikey=settings.CKAN_API_KEY)
+
+        packages = remote.action.package_search(
+            fq='tags:site-images',
+            include_private=True,
+            rows=100000
+        )['results']
+
+        logger.info('There are {} image packages.'.format(len(packages)))
+
+        lookup_table = defaultdict(list)
+
+        for i in packages:
+            try:
+                coords = (i['latitude'], i['longitude'])
+                img_url = i['resources'][0]['url']
+
+                lookup_table[coords].append(img_url)
+
+            except Exception as e:
+                logger.error("Either latitude or longitude is missing for: {}".format(img_url))
+                logger.error("Missing parameter is: {}".format(str(e)))
+
+                continue
+
+        lookup_table = dict(lookup_table)
+        _set_cached_item(LOOKUP_TABLE_KEY, lookup_table)
+
+
+def process_img(request=None, lat=None, lng=None, index=None):
+    '''
+    Return specified cached image or fetch image from ckan and resize before caching and returning.
+    '''
+    _create_img_lookup_table()
+
+    lookup_table = _get_cached_item(LOOKUP_TABLE_KEY)
```

Comment:
I think this would be better as:

`lookup_table = get_lookup_table()`

and then hide the caching stuff in there, rather than having a function `create_img_lookup_table()` which is a precondition to getting this cache key, but may or may not mutate global state

Reply:
Fixed.